run.sh

```diff
@@ -5,6 +5,9 @@ set -e
 # this will create venv from python version defined in .python-version
 uv venv
 
+# Activate the virtual environment using the dot command
+. .venv/bin/activate
+
 # install requirements for the project
 uv pip install -r requirements.txt
```

Comment:
works well!  once we activate venv, we don't need to call "uv run", instead just do the following & deactivate

```sh
# run app using python from venv
echo "Running Ring with $(python3 --version) at '$(which python3)'"
python3 main.py

# deactivate the virtual environment
deactivate
```

Reply:
Thanks @yashgorana. This works nicely
![Screenshot from 2024-11-05 09-24-41](https://github.com/user-attachments/assets/9dc00506-46b8-4a5e-a86b-6c25cf2f2909)